mysys_ssl/my_aes.cc

```diff
@@ -57,3 +57,99 @@ void my_aes_create_key(const unsigned char *key, uint key_length,
     *ptr^= *sptr;
   }
 }
+
+int my_aes_encrypt_nopad(const unsigned char *source, uint32 source_length,
+                         unsigned char *dest,
+                         const unsigned char *key, uint32 key_length,
+                         enum my_aes_opmode mode, const unsigned char *iv)
+{
+  unsigned char eiv[MY_AES_BLOCK_SIZE];
+  uint32 aligned_src_len= (source_length / MY_AES_BLOCK_SIZE)
+                          * MY_AES_BLOCK_SIZE;
+  enum my_aes_opmode ecb_mode;
+  int dst_len;
+
+  if (key_length == 64)
+    ecb_mode= my_aes_256_ecb;
+  else if (key_length == 32)
+    ecb_mode= my_aes_128_ecb;
+  else
+    return MY_AES_BAD_DATA;
+
+  my_aes_encrypt(iv, MY_AES_BLOCK_SIZE, eiv,
+                 key, key_length, ecb_mode, NULL, false);
+
+  dst_len= my_aes_encrypt(source, aligned_src_len, dest,
+                          key, key_length, mode, eiv, false);
+  if (dst_len == MY_AES_BAD_DATA)
+    return dst_len;
+
+  DBUG_ASSERT(dst_len == static_cast<int>(aligned_src_len));
+
+  if (aligned_src_len < source_length)
+  {
+    unsigned char mask[MY_AES_BLOCK_SIZE];
+    uint32 len= source_length - aligned_src_len;
+
+    dst_len= my_aes_encrypt(eiv, MY_AES_BLOCK_SIZE, mask,
+                            key, key_length, ecb_mode, NULL, false);
+
+    if (dst_len == MY_AES_BAD_DATA)
+      return dst_len;
+
+    for (uint32 i= 0; i < len; i++)
+      dest[aligned_src_len + i]= source[aligned_src_len + i] ^ mask[i];
+
+    return source_length;
+  }
+
+  return source_length;
+}
+
+int my_aes_decrypt_nopad(const unsigned char *source, uint32 source_length,
+                         unsigned char *dest,
+                         const unsigned char *key, uint32 key_length,
+                         enum my_aes_opmode mode, const unsigned char *iv)
+{
+  unsigned char eiv[MY_AES_BLOCK_SIZE];
+  uint32 aligned_src_len= (source_length / MY_AES_BLOCK_SIZE)
+                          * MY_AES_BLOCK_SIZE;
+  enum my_aes_opmode ecb_mode;
+  int dst_len;
+
+  if (key_length == 64)
+    ecb_mode= my_aes_256_ecb;
+  else if (key_length == 32)
+    ecb_mode= my_aes_128_ecb;
```

Comment:
This seems to be wrong. In AES 128 and 256 mean key sizes. Thus key length of 64 bytes means aes 512 and key length of 32 bytes means aes 256.

Reply:
Good catch. Fixed